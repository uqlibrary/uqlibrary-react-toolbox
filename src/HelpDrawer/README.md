# HelpDrawer

simple content menu for UQ Library applications using ReactJs

## Props
- `title: React.PropTypes.string`
- `text: React.PropTypes.any.isRequired` 
- `buttonLabel: React.PropTypes.any.isRequired`
- `displayType: React.PropTypes.bool` - currently only inline
- `style: React.PropTypes.object`
    
## Usage

**./App.js**
```jsx
import React from 'react';
...

import {HelpDrawer} from 'uqlibrary-react-toolbox';

// Data configuration
const text = 'Where ever additional cues or explanation are required to clarify a process or procedure. Can be used as a card cue (inline inside of <CardHeader> to offer the icon in the top right of the card, or inline in text or form elements. Additionally, by adding 2 parameters (helpTitle and helpText) to a <Field> element, an integrated help icon can be produced. More info on this implementation in forms.'
        
const App = () => (
  <div>
    <HelpIcon 
        title="Contextual help drawer" 
        text={text} 
        buttonLabel="Got it!" 
        inline />
  </div>
);

ReactDOM.render(
  <App />,
  document.getElementById('app')
);
```

