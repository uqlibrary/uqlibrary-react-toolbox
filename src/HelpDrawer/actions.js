// Types
export const SHOW = 'HELP_DRAWER_SHOW';
export const HIDE = 'HELP_DRAWER_HIDE';

// Actions
export function show(title, text, buttonLabel) {
    return {
        type: SHOW,
        payload: {title, text, buttonLabel}
    };
}

export function hide() {
    return {
        type: HIDE
    };
}
