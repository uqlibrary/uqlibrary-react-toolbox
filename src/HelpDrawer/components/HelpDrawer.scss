@import '../../sass/variables';

.help-drawer {
  .app-bar {
    height: 75px;
    background-color: $theme-color;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    width: 100%;
  }

  .content {
    padding: 30px;
    overflow-y: auto;
  }
}
