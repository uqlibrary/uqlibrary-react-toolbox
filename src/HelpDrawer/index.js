export {default as HelpIcon} from './containers/HelpIcon';
export {default as HelpDrawer} from './containers/HelpDrawer';

export {default as helpDrawerReducer} from './reducer';
