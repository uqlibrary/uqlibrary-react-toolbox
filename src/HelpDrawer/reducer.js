import Immutable from 'immutable';

import {
    HIDE,
    SHOW
} from './actions';

const initialState = Immutable.fromJS({
    open: false,
    title: '',
    text: '',
    buttonLabel: 'OK'
});

const helpDrawerReducer = (state = initialState, action) => {
    switch (action.type) {
        case SHOW:
            return Immutable.fromJS({...action.payload, open: true});
        case HIDE:
            return initialState;
        default:
            return state;
    }
};

export default helpDrawerReducer;
