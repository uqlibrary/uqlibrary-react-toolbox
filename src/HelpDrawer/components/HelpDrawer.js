import React from 'react';
import Drawer from 'material-ui/Drawer';
import RaisedButton from 'material-ui/RaisedButton';
import Divider from 'material-ui/Divider';

import './HelpDrawer.scss';

const propTypes = {
    open: React.PropTypes.bool.isRequired,
    title: React.PropTypes.string.isRequired,
    text: React.PropTypes.any.isRequired,
    hide: React.PropTypes.func.isRequired,
    buttonLabel: React.PropTypes.string
};

const defaultProps = {
    buttonLabel: 'Ok'
};

const HelpDrawer = ({title, text, buttonLabel, open, hide}) => {
    const toggleDrawer = () => {
        hide();
    };

    return (
        <Drawer
            containerClassName="help-drawer flex"
            open={open}
            openSecondary
            docked={false}
            disableSwipeToOpen
            width={380}
            onRequestChange={toggleDrawer}>
            <div className="layout-fill side-drawer column align-stretch">
                <div className="flex column content">
                    <h1 className="headline">{title}</h1>
                    <Divider />
                    <div className="subhead">{text}</div>
                </div>
                <div className="row justify-end layout-padding">
                    <RaisedButton secondary label={buttonLabel} onTouchTap={toggleDrawer} />
                </div>
            </div>
        </Drawer>
    );
};

HelpDrawer.propTypes = propTypes;
HelpDrawer.defaultProps = defaultProps;

export default HelpDrawer;
