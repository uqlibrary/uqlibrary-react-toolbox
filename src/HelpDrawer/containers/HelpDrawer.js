import {connect} from 'react-redux';

import {hide} from '../actions';
import HelpDrawer from '../components/HelpDrawer';

const HelpDrawerContainer = connect(state => {
    return state.get('helpDrawer').toJS();
}, dispatch => {
    return {
        hide: () => dispatch(hide())
    };
})(HelpDrawer);

export default HelpDrawerContainer;
