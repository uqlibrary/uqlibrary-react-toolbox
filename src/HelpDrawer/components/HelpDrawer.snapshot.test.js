jest.dontMock('./HelpDrawer');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import HelpDrawer from './HelpDrawer';

function setup(title, text, buttonLabel, open) {
    const props = {
        open: open,
        title: title,
        text: text,
        hide: () => {},
        buttonLabel: buttonLabel
    };

    return shallow(<HelpDrawer {...props} />);
}

describe('HelpDrawer snapshots tests', () => {
    it('renders menu', () => {
        const hdText = 'Integer mattis rutrum velit nec posuere. Quisque rhoncus quam elit, eu tincidunt diam feugiat eu. Quisque luctus luctus mauris faucibus ornare. Ut eu metus vitae est euismod gravida ac vel augue. Duis sapien massa, tempor id vulputate nec, auctor nec augue. Donec ullamcorper dignissim metus at vulputate. Mauris non magna enim. Quisque gravida libero augue, efficitur vestibulum leo porttitor id. Etiam quis nisi vehicula, eleifend turpis ut, accumsan nisi. Nam nec nibh auctor, placerat ex vel, rhoncus risus. In fermentum ex sit amet augue molestie sodales. Sed eleifend convallis dui id euismod. Nam eu turpis non mi facilisis euismod. Morbi congue et lectus sed tempus. Interdum et malesuada fames ac ante ipsum primis in faucibus.';

        const wrapper = setup('HelpDrawer Title', hdText, 'Close', true);
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
