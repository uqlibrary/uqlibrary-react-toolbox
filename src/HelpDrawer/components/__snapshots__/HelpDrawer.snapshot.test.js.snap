// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpDrawer snapshots tests renders menu 1`] = `
<Drawer
  containerClassName="help-drawer flex"
  disableSwipeToOpen={true}
  docked={false}
  onRequestChange={[Function]}
  open={true}
  openSecondary={true}
  swipeAreaWidth={30}
  width={380}
  zDepth={2}
>
  <div
    className="layout-fill side-drawer column align-stretch"
  >
    <div
      className="flex column content"
    >
      <h1
        className="headline"
      >
        HelpDrawer Title
      </h1>
      <Divider
        inset={false}
      />
      <div
        className="subhead"
      >
        Integer mattis rutrum velit nec posuere. Quisque rhoncus quam elit, eu tincidunt diam feugiat eu. Quisque luctus luctus mauris faucibus ornare. Ut eu metus vitae est euismod gravida ac vel augue. Duis sapien massa, tempor id vulputate nec, auctor nec augue. Donec ullamcorper dignissim metus at vulputate. Mauris non magna enim. Quisque gravida libero augue, efficitur vestibulum leo porttitor id. Etiam quis nisi vehicula, eleifend turpis ut, accumsan nisi. Nam nec nibh auctor, placerat ex vel, rhoncus risus. In fermentum ex sit amet augue molestie sodales. Sed eleifend convallis dui id euismod. Nam eu turpis non mi facilisis euismod. Morbi congue et lectus sed tempus. Interdum et malesuada fames ac ante ipsum primis in faucibus.
      </div>
    </div>
    <div
      className="row justify-end layout-padding"
    >
      <RaisedButton
        disabled={false}
        fullWidth={false}
        label="Close"
        labelPosition="after"
        onTouchTap={[Function]}
        primary={false}
        secondary={true}
      />
    </div>
  </div>
</Drawer>
`;
